### PR TITLE
Fix initialization issues in KieServer*ControllerClient testing clients

### PR DESCRIPTION
**Pull request by jhrcek**
I'm trying to reuse this testing rest client in our selenium test suites to save some some clicking in UI.
While using these we came across 2 issues:
  * custom credentials do not work
  * marshaller is not initialized properly

@sutaakar Could you please review?

### CODE REVIEW

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/controller/client/KieServerControllerClient.java [hunk 1]
```diff
@@ -28,25 +28,35 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * @depracated Use {@link KieServerMgmtControllerClient} instead.
+ */
```
Comment (triceo) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/controller/client/KieServerControllerClient.java]: Why not `@Deprecated`?

Comment (jhrcek) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-common/src/main/java/org/kie/server/integrationtests/controller/client/KieServerControllerClient.java]: Why not indeed? Updated PR coming..

**Comment by sutaakar** [comment]
Looks good.

**Comment by jhrcek** [comment]
Hello @mswiderski, could you please review/merge this? I need these changes to be able to run selenium jbpm-console-ng tests on jenkins..

**Comment by mswiderski** [comment]
+1

**Comment by mswiderski** [comment]
do we need it in 6.5.x?

**Comment by jhrcek** [comment]
No, it's just for testing kie-wb after jbpm runtime removal. Thank you!

**Pull request merged**